tools/ci/markdown-link-verificator.sh

```diff
@@ -0,0 +1,312 @@
+        if [[ $(echo $LINK | grep "https://github.com/eclipse-iceoryx/iceoryx/tree/" | wc -l ) == "1" ]]
+        then
+            local PATH_AFTER_TREE=$(echo $LINK | sed -n "s/https:\/\/github.com\/eclipse-iceoryx\/iceoryx\/tree\/\(.*\)/\1/p")
+            if [[ $(echo $PATH_AFTER_TREE | grep "/" | wc -l ) == "1" ]]
+            then
+                echo -e "${COLOR_LIGHT_RED}Please do not use a github url when also a relative path works!${COLOR_RESET}"
+                echo -e "${COLOR_LIGHT_RED}Relative paths ensure that the tag/branch of link source and destination is equal.${COLOR_RESET}"
+                printLinkFailureSource
+                return
+            fi
+        fi
+
+        # verify that no link to a specific github branch is used in the documentation
+        if [[ $(echo $LINK | grep "https://github.com/eclipse-iceoryx/iceoryx/blob" | wc -l) == "1" ]]
+        then
+            echo -e "${COLOR_LIGHT_RED}Please do not use a github url when also a relative path works!${COLOR_RESET}"
+            echo -e "${COLOR_LIGHT_RED}Relative paths ensure that the tag/branch of link source and destination is equal.${COLOR_RESET}"
+            printLinkFailureSource
+            return
+        fi
+
+        if [[ $(echo $LINK | grep "https://iceoryx.io" | wc -l ) == "1" ]]
+        then
+            echo -e "${COLOR_LIGHT_RED}Please try to avoid to link directly to the documentation on https://iceoryx.io!${COLOR_RESET}"
+            echo -e "${COLOR_LIGHT_RED}This causes switching between multiple sources of the same material.${COLOR_RESET}"
+            printLinkFailureSource
+            return
+        fi
+
+        if ! [[ $(doesWebURLExist $LINK) == "1" ]]
+        then
+            printLinkFailureSource
+        fi
+    fi
+}
+
+verifyLinkToFile()
+{
+    LINK_VALUE=$1
+
+    if [[ $(isAbsolutePath $LINK_VALUE) == "1" ]]
+    then
+        LINK=${ICEORYX_ROOT_PATH}/${LINK_VALUE}
+    else
+        LINK=${FILE_DIRECTORY}/${LINK_VALUE}
+    fi
+
+    if [[ $(echo $LINK | grep '#' | wc -l) == "1" ]]
+    then
+        SECTION_IN_FILE=$(echo $LINK | cut -f 2 -d '#')
+        LINK=$(echo $LINK | cut -f 1 -d '#')
+    fi
+
+    if ! [ -f $LINK ] && ! [ -d $LINK ]
+    then
+        printLinkFailureSource
+
+        POSSIBLE_ALTERNATIVE=$(find $ICEORYX_ROOT_PATH -type f -iname $(basename $LINK))
+        echo -e "Is this the file you are looking for: ${COLOR_LIGHT_BLUE}$POSSIBLE_ALTERNATIVE${COLOR_RESET}"
+        echo
+        SECTION_IN_FILE=""
+        return
+    fi
+
+    if ! [[ $SECTION_IN_FILE == "" ]]
+    then
+        verifyLinkToSection "#$SECTION_IN_FILE" $LINK
+    fi
+
+    SECTION_IN_FILE=""
+}
+
+checkLinksInFile()
+{
+    FILE=$1
+
+    FILE_DIRECTORY=$(dirname $FILE)
+    IS_IN_CODE_ENV="0"
+
+    readarray FILE_CONTENT < $FILE
+    LINE_NR=0
+    for LINE in "${FILE_CONTENT[@]}"
+    do
+        let LINE_NR=$LINE_NR+1
+
+        # detect code environments, see ``` and skip them
```

Comment:
there can be more than three `, e.g. you have a code block which shows a code block
``````
````
see this code
```
all glory to the hypnotoad
```
````
``````
The important part is that the number of ` from the block opening must match the block closing

Reply:
I didn't know that this was even possible ...

Reply:
Would it be alright with you to support for the moment only code blocks with 3 ` and print a warning when 4 ` are found?

We do not have this anywhere in the code so I think we should be safe for now and when the use case arises we get a meaningful error message and can implement it.